### PR TITLE
Library: Rename "object safe" to "dyn compatible"

### PR DESCRIPTION
**Pull request by fmease**
Completed T-lang FCP: https://github.com/rust-lang/lang-team/issues/286#issuecomment-2338905118.
Tracking issue: https://github.com/rust-lang/rust/issues/130852

Regarding https://github.com/rust-lang/rust/labels/relnotes, I guess I will manually open a https://github.com/rust-lang/rust/labels/relnotes-tracking-issue since this change affects everything (compiler, library, tools, docs, books, everyday language).

r? ghost

### CODE REVIEW

File: library/core/src/ops/unsize.rs [hunk 1]
```diff
@@ -112,6 +112,9 @@ impl<T: ?Sized + Unsize<U>, U: ?Sized> CoerceUnsized<*const U> for *const T {}
 ///     T: Unsize<U>,
 /// {}
 /// ```
+///
+/// [^1]: Formerly known as *object safety*.
+/// [^2]: Formerly known as *object safe*.
```
Comment (fmease) [inline library/core/src/ops/unsize.rs:117]: awkward duplication

**Comment by fmease** [comment]
r? libs (or libs-api?)

File: library/core/src/ops/unsize.rs (hunk 1 above)
Comment (ibraheemdev) [reply library/core/src/ops/unsize.rs:117]: It seems reasonable to remove second footnote given that dyn-incompatiblity was previously defined.

Comment (fmease) [reply library/core/src/ops/unsize.rs:117]: Done

**Pull request merged**